src/main/java/fruitshop/service/writer/FileWriter.java

```
@@ -0,0 +1,7 @@
+package fruitshop.service.writer;
+
+import java.io.IOException;
+
+interface FileWriter {
+    void write(String data, String fileName) throws IOException;
```

This violates checklist item: 'Don't ignore exceptions in try-catch blocks'. The method signature throws IOException, but the requirements specify that IOExceptions should be caught and rethrown as RuntimeExceptions with informative messages and the original cause.